Documentation/ApiOverview/RequestHandling/Index.rst

```diff
@@ -300,6 +300,149 @@ identifier. This will circumvent the risk of circularity::
    Always check the integrity of the middleware stack after changing the default ordering.
    This can be done in the configuration module that comes with EXT:lowlevel.
 
+.. _request-handling-psr-17:
+
+Creating new Request / Response Objects
+=======================================
+
+PSR-17_ HTTP Factory interfaces are provided by `psr/http-factory` and should be used as
+dependencies for PSR-15_ request handlers or services that need to create PSR-7_ message objects.
+
+It is discouraged to explicitly create PSR-7_ instances of classes from the :php:`\TYPO3\CMS\Core\Http`
+namespace (they are not public API). Use type declarations against PSR-17_ HTTP Message Factory
+interfaces and dependency injection instead.
```

Comment:
```suggestion
namespace (they are not public APIs). Instead, use type declarations against PSR-17_ HTTP Message Factory
interfaces and dependency injections.
```
I could be wrong. Unsure about what is singular and plural here.

Reply:
"Dependency Injection" should always be singular as it describes the pattern / technology and is kind of a fixed term.

Reply:
Oh right, cool.